nutils/function.py

```diff
@@ -3789,6 +3789,22 @@ def __init__(self, *, default_geometry_name='x'):
     super().__setattr__('default_geometry_name', default_geometry_name)
     super().__init__()
 
+  def __getstate__(self):
+    def catchgetattr(attr):
+      cls = self.__class__
+      try:
+        return attr, getattr(self, attr)
+      except AttributeError as e:
+        import inspect
+        log.warning('{}.{}.{} failed to add {}'.format(cls.__module__,
+          cls.__name__, inspect.stack()[2].function, attr))
+        return attr, None
+    attrs = '_arg_shapes', '_attributes', 'default_geometry_name'
+    return {k: v for k, v in map(catchgetattr, attrs) if v is not None}
+
```

Comment:
Is `catchgetattr` really necessary? The attributes to be pickled are set in `__init__`, so I don't see why `getattr` might fail.

Reply:
Absolutely right.